redhat_access_insights/connection.py

```diff
@@ -169,6 +153,13 @@ def get_proxies(self, config):
                 conf_proxy = scheme + location
             logger.debug("CONF Proxy: %s", conf_proxy)
             proxies = {"https": conf_proxy}
+
+        if os.environ.get('NO_PROXY') and conf_proxy:
+            logger.debug("Invalid Proxy configuration. "
+                         "You have environment variable NO_PROXY set "
+                         "as well as 'proxy' set in your configuration file. "
+                         "This is invalid. Do not expect normal results.")
```

Comment:
I agree that you could just eliminate the conf file setting. I bet it
was put there to help with configuration for puppet/chef/etc (not that
those cant set their own variables) and to make registration via a
proxy persistent. Think hard or rather look at hard the issues with
the cron job that might come up if you switch to removing the conf
file completely. Keep in mind crond jobs often do not run with the
full environment variables you set elsewhere.

Also saying "We certainly don't want to start setting environment
variables in our code." is kinda funny. When in python you "set" and
environment variable you are only affecting yourself... or the running
processes env. That stuff is not persisted after the python
interpreter exits. You are not "polluting" anything.

I would leave the conf file setting, have it modify the shell env,
letting requests take care, and do the right thing...
Or remove the conf file setting all together...
I would not do the patch that complains about an unsupported config.

That being said, I would lean towards the first option. Just sayin.

On Thu, Aug 18, 2016 at 4:56 PM, Richard Brantley
<notifications@github.com> wrote:
> In redhat_access_insights/connection.py:
>
>> @@ -169,6 +153,13 @@ def get_proxies(self, config):
>>                  conf_proxy = scheme + location
>>              logger.debug("CONF Proxy: %s", conf_proxy)
>>              proxies = {"https": conf_proxy}
>> +
>> +        if os.environ.get('NO_PROXY') and conf_proxy:
>> +            logger.debug("Invalid Proxy configuration. "
>> +                         "You have environment variable NO_PROXY set "
>> +                         "as well as 'proxy' set in your configuration
>> file. "
>> +                         "This is invalid. Do not expect normal
>> results.")
>
> I don't think we need to have a proxy setting in the conf file at all. It
> should be handled via HTTPS_PROXY and NO_PROXY. This would ensure that
> everything is handled transparently behind the scenes by urllib/requests and
> the OS level, therefore we need not worry about any logic on our end.
> Although, the logic still has to be in there now at this point to maintain
> backwards compatibility, however moving foward we can say we do not support
> proxy config file variables and we only support HTTPS_PROXY or NO_PROXY
> environment variable configurations. We certainly don't want to start
> setting environment variables in our code.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.



-- 
Sent from me, iHands.